### PR TITLE
Add tests for ignoring VDIs when snapshotting a VM

### PR DESCRIPTION
**Pull request by benjamreis**
See: https://github.com/xapi-project/xen-api/pull/4563

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

### CODE REVIEW

**Comment by benjamreis** [comment]
Do NOT merge until the upstream PR is merged as some small things might change.

File: tests/snapshot/test_snapshot_ignore_vdis.py
Comment (stormi) [inline tests/snapshot/test_snapshot_ignore_vdis.py:22]: Looks like this whole part could be extracted to a function

File: lib/common.py
Comment (Wescoeur) [inline lib/common.py:774]: Maybe `ignored_vdis`. Otherwise I have the impression that the param is a boolean.

File: tests/snapshot/conftest.py
Comment (Wescoeur) [inline tests/snapshot/conftest.py:11]: Too many `C-c` & `C-v`, you can use instead:

```python
make_vdi = lambda name: host.xe('vdi-create', {'name-label': name, 'virtual-size': '64', 'sr-uuid': sr.uuid})
vdi_A, vdi_B, vdi_C = make_vdi('VDI_A'), make_vdi('VDI_B'), make_vdi('VDI_C') # Or using multi-lines. ;)
```

Comment (Wescoeur) [reply tests/snapshot/conftest.py:11]: Note: do what you want, but we should think about it for more complex tests. :)

File: tests/snapshot/test_snapshot_ignore_vdis.py
Comment (Wescoeur) [inline tests/snapshot/test_snapshot_ignore_vdis.py:42]: Maybe it would be useful to have a helper function to do that. :)

File: lib/common.py
Comment (benjamreis) [reply lib/common.py:774]: This is to reflect the call in xapi (the param is called `ignore_vdis` & `ignore-vdi-uuids`)

**Review by stormi — APPROVED** [review]

**Review by Wescoeur — APPROVED** [review]

**Pull request merged**
